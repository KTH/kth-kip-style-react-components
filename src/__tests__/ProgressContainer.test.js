import React from 'react'
import renderer from 'react-test-renderer'
import ProgressBar from '../lib/ProgressContainer'

const TEST_DATA = {
  pages: [
    {
      title: '1. Välj bild',
      intro: `Choose a semester and a course offering for the course memo data to be published (step 1 of 3).
      In the next step (2 of 3), course data will be fetched automatically for the selected semester and course offering.
      It is there possible to edit some of the course data and course memo data.
      Preview the table with the course data and course memo data that are about to be published in the last step (3 of 3).
      The course data with the course memo data will then be published on the page ....
      `
    },
    {
      title: '2. Redigera text',
      intro: `In this step (2 of 3) course memo data and course memo shall be uploaded,
      changes to the chosen course offering is summarized and some of the course data are reviewed and adjusted.`
    },
    {
      title: '3. Granska och publicera',
      intro: `Börja med att välja vilken bild som ska visas på kursinformationssidan (steg 1 av 3). I nästa steg (2 av 3) kommer du att kunna redigera den inledande texten. 
      I sista steget (3 av 3) ges möjlighet att först granska bild och text och sedan publicera det på sidan ”Kursinformation”.`
    }
  ]
}

describe('ProgressContainer', () => {
  it('renders properly', () => {
    const tree = renderer
      .create(<ProgressBar active={1} pages={TEST_DATA.pages} />
        ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
