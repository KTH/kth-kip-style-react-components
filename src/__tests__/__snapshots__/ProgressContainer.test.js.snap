// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressContainer renders properly 1`] = `
<div
  className="progress-bar-container row"
>
  <div
    className="col-md-4 col-sm-12 progress-bar1 progress-active"
    style={
      Object {
        "borderBottom": "2px solid #d4d4d4",
        "fontWeight": "900",
        "marginBottom": "10px",
        "paddingLeft": "0",
      }
    }
  >
    <span>
      1. VÃ¤lj bild
    </span>
  </div>
  <div
    className="col-md-4 col-sm-12 progress-bar1 "
    style={
      Object {
        "borderBottom": "2px solid #d4d4d4",
        "marginBottom": "10px",
        "paddingLeft": "0",
      }
    }
  >
    <span>
      2. Redigera text
    </span>
  </div>
  <div
    className="col-md-4 col-sm-12 progress-bar1 "
    style={
      Object {
        "borderBottom": "2px solid #d4d4d4",
        "marginBottom": "10px",
        "paddingLeft": "0",
      }
    }
  >
    <span>
      3. Granska och publicera
    </span>
  </div>
  <div>
    <p
      data-testid="intro-text"
      style={
        Object {
          "marginBottom": "0",
          "marginTop": "20px",
        }
      }
    >
      Choose a semester and a course offering for the course memo data to be published (step 1 of 3).
      In the next step (2 of 3), course data will be fetched automatically for the selected semester and course offering.
      It is there possible to edit some of the course data and course memo data.
      Preview the table with the course data and course memo data that are about to be published in the last step (3 of 3).
      The course data with the course memo data will then be published on the page ....
    </p>
  </div>
</div>
`;
